### PR TITLE
Creating new games

### PR DESCRIPTION
**Pull request by mtchavez**
[Agilatr#40] - https://www.agilatr.com/projects/bm5k-meepl-es/stories/40

* Creating new games
  * Add create/new routes for games
  * Add migration for added_by to games
  * New game form
  * Creating game feature specs

### CODE REVIEW

**Comment by mtchavez** [comment]
Need to know where the link to create new games should go then I can add that and stage.

**Comment by BM5k** [comment]
@mtchavez on the search results page?

**Comment by mtchavez** [comment]
Sure. @bowerman you good with that?

**Comment by mtchavez** [comment]
Staged

**Comment by BM5k** [comment]
https://travis-ci.org/meeples/meeples/builds/53500502#L286-L289

**Comment by BM5k** [comment]
@mtchavez the test failures https://travis-ci.org/meeples/meeples/builds/53500502#L485-L498 actually look legit

**Comment by mtchavez** [comment]
Yeah. My bad. I made a change and pushed it to staging without reviewing the failures.

**Comment by bowerman** [comment]
@mtchavez yeah, I'm alright with it.

**Comment by mtchavez** [comment]
Fixed and staged again.

**Comment by BM5k** [comment]
See my notes in [agilatr][1], and this here screenshot:

screen shot 2015-03-15 at 6 01 42 pm

[1]: https://www.agilatr.com/projects/bm5k-meepl-es/stories/40#comment_904

**Comment by mtchavez** [comment]
Can we not split comments across technologies in the future?

* Do we want year published to be optional?

**Comment by BM5k** [comment]
@mtchavez yeah, my bad

File: app/views/games/new.html.haml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  %h2
+    New Game
+
+.row-fluid.col-sm-12
+  = simple_form_for @game do |f|
+    = f.input :name
+    .form-group.string.optional.game_year_published
+      = f.label :year_published, 'Year Published'
+      %div
+        = f.select :year_published, options_for_select((1901..Date.today.year), selected: nil), include_blank: true
```
Comment (bowerman) [inline app/views/games/new.html.haml]: rails has a select_year date helper that may be more readable.

File: spec/features/games_spec.rb [hunk 2]
```diff
@@ -41,6 +41,28 @@
...
+
+    before do
+      visit new_game_path
+    end
+
+    it 'requires unique name' do
+      click_button 'Add'
+      should have_content 'Game was not successfully created!'
+      select Time.now.year, from: 'game_year_published'
+      should have_content "Namecan't be blank"
```
Comment (bowerman) [inline spec/features/games_spec.rb]: not sure if this was intentional

File: app/views/games/new.html.haml (hunk 1 above)
Comment (mtchavez) [inline app/views/games/new.html.haml]: It created some weird nested form stuff like `params: {'date': {'year': 2015}}`

File: spec/features/games_spec.rb (hunk 2 above)
Comment (mtchavez) [inline spec/features/games_spec.rb]: If you mean the white space, no. Will add between expectations for readability.

Comment (bowerman) [inline spec/features/games_spec.rb]: I'm not sure what you mean, I'm talking about `Namecan't`

File: app/views/games/new.html.haml (hunk 1 above)
Comment (BM5k) [inline app/views/games/new.html.haml]: Just as an FYI, `Date.current` is time zone aware, and `Date.today` isn't.

File: spec/features/games_spec.rb (hunk 2 above)
Comment (mtchavez) [inline spec/features/games_spec.rb]: Oh, I can't tell what you were asking in that original comment. I assumed it was the white space stuff. My bad. It's what shows up in the raw content for some reason.

File: spec/features/games_spec.rb [hunk 3]
```diff
@@ -41,6 +41,28 @@
...
+  describe 'creating' do
+
+    before do
+      visit new_game_path
+    end
+
+    it 'requires unique name' do
+      click_button 'Add'
+      should have_content 'Game was not successfully created!'
+      select Time.now.year, from: 'game_year_published'
```
Comment (BM5k) [inline spec/features/games_spec.rb]: this line doesn't affect the assertion made on the next line

**Pull request merged**